ui/components/src/ColumnsEditor/ColumnEditorContainer.tsx

```diff
@@ -0,0 +1,77 @@
+// Copyright 2024 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { IconButton, Stack, Typography } from '@mui/material';
+import ChevronRight from 'mdi-material-ui/ChevronRight';
+import ChevronDown from 'mdi-material-ui/ChevronDown';
+import DeleteIcon from 'mdi-material-ui/DeleteOutline';
+import EyeIcon from 'mdi-material-ui/EyeOutline';
+import EyeOffIcon from 'mdi-material-ui/EyeOffOutline';
+import { ColumnEditor, ColumnEditorProps } from './ColumnEditor';
+
+export interface ColumnEditorContainerProps extends ColumnEditorProps {
+  isCollapsed: boolean;
+  onCollapse: (collapsed: boolean) => void;
+  onDelete: () => void;
+}
+
+export function ColumnEditorContainer({
+  column,
+  isCollapsed,
+  onChange,
+  onCollapse,
+  onDelete,
+}: ColumnEditorContainerProps) {
+  function handleHideColumn() {
+    onChange({ ...column, hide: !column.hide });
+  }
+
+  return (
+    <Stack spacing={1}>
+      <Stack
+        direction="row"
+        alignItems="center"
+        borderBottom={1}
+        borderColor={(theme) => theme.palette.divider}
+        justifyContent="space-between"
+        gap={4}
+      >
+        <Stack direction="row" gap={1}>
+          <IconButton size="small" onClick={() => onCollapse(!isCollapsed)}>
+            {isCollapsed ? <ChevronRight /> : <ChevronDown />}
+          </IconButton>
+          <Typography variant="overline" component="h4" sx={{ textTransform: 'none' }}>
+            COLUMN:{' '}
+            {column.header ? (
+              <span>
+                <strong>{column.header}</strong> ({column.name})
+              </span>
+            ) : (
+              <strong>{column.name}</strong>
+            )}
+          </Typography>
+        </Stack>
+
+        <Stack direction="row" gap={1}>
+          <IconButton size="small" sx={{ marginLeft: 'auto' }} onClick={handleHideColumn}>
+            {column.hide ? <EyeOffIcon /> : <EyeIcon />}
```

Comment:
As already discussed personally I'm not fan of this show/hide button being at the top-right
![image](https://github.com/user-attachments/assets/49abd78b-9e3b-4a2b-8ffe-de142430dfbe)

To me it goes against an implicit convention here,
- the trashbin icon seems to mean "remove this column settings" and that's actually what it does: it removes the column settings, not the column itself in the generated table.
- following the same logic, the eye icon seems to mean "hide this column settings", which doesn't make sense, and it actually hides the column in the generated table. 

Thus I'd rather have a show/hide toggle within the column settings part (in first or second position, as if you hide a column you don't care about providing descriptions/headers that will never get displayed)

I know the main advantage to have this eye icon there is to have access to this show/hide feature even if the corresponding column settings is collapsed:
![image](https://github.com/user-attachments/assets/bd27cf40-57d8-4f34-b9f9-acd7d8c5f4fc)

Still, I value more UX consistency than saving clicks here, so I still prefer my proposed design 🤓 

But I guess we could manage to find a solution satisfying both, e.g move the controls to the row when it's collapsed / to the form when it's expanded? I can't come up with a design idea right now..

Reply:
I will replicate it in the editor

Reply:
> But I guess we could manage to find a solution satisfying both, e.g move the controls to the row when it's collapsed / to the form when it's expanded? I can't come up with a design idea right now..

Just for context, a draft of what I had in mind (collapsed vs expanded view)

![Propal table](https://github.com/user-attachments/assets/43069cbd-b4aa-4439-8927-4358b7920e7c)

Reply:
I'm fine with your latest change (the eye icon is hidden when the row is expanded) as a quickwin mitigation, let's go with this for now.